endpointmanager/pkg/historypruning/historypruning_integration_test.go

```diff
@@ -0,0 +1,795 @@
+
+	code := m.Run()
+
+	teardown()
+	os.Exit(code)
+}
+
+func Test_PruneInfoHistory(t *testing.T) {
+	teardown, _ := th.IntegrationDBTestSetup(t, store.DB)
+	defer teardown(t, store.DB)
+
+	setupCapabilityStatement(t, filepath.Join("../testdata", "cerner_capability_dstu2.json"))
+
+	ctx := context.Background()
+
+	var err error
+
+	addFHIREndpointInfoHistoryStatement, err = store.DB.Prepare(`
+	INSERT INTO fhir_endpoints_info_history (
+		operation, 
+		entered_at, 
+		id, 
+		url,
+		tls_version,
+		mime_types,
+		smart_response, 
+		capability_statement)			
+	VALUES ($1, $2, $3, $4, $5, $6, $7, $8);`)
+	th.Assert(t, err == nil, err)
+	defer addFHIREndpointInfoHistoryStatement.Close()
+
+	var historyURL = "http://example.com/DTSU2/"
+	// reset context
+	ctx = context.Background()
+
+	// Add few days to the threshold to make sure date is older than a month
+	threshold := viper.GetInt("pruning_threshold")
+	pastDate := threshold + 3*(1440)
+	var idExpectedArr []int
+	var checkCorrectness bool
+
+	var idActual int
+	getIDStatement, err = store.DB.Prepare(`SELECT id FROM fhir_endpoints_info_history WHERE url = $1 ORDER BY entered_at ASC;`)
+	th.Assert(t, err == nil, err)
+	defer getIDStatement.Close()
+
+	var count int
+	ctStatement, err := store.DB.Prepare(`SELECT count(*) FROM fhir_endpoints_info_history WHERE url = $1;`)
+	th.Assert(t, err == nil, err)
+	defer ctStatement.Close()
+
+	// Clear history table in database
+	clearStatement, err := store.DB.Prepare(`DELETE FROM fhir_endpoints_info_history WHERE url = $1;`)
+	th.Assert(t, err == nil, err)
+	defer clearStatement.Close()
+	_, err = clearStatement.ExecContext(ctx, historyURL)
+	th.Assert(t, err == nil, err)
+
+	// Add two endpoint entries to info history table with current entered_at dates
+	err = AddFHIREndpointInfoHistory(ctx, store, testFhirEndpointInfo, time.Now().Format("2006-01-02 15:04:05.000000000"), idCount, "U")
+	th.Assert(t, err == nil, err)
+
+	err = AddFHIREndpointInfoHistory(ctx, store, testFhirEndpointInfo, time.Now().Format("2006-01-02 15:04:05.000000000"), idCount, "U")
+	th.Assert(t, err == nil, err)
+
+	// Ensure both entries were added to info history table correctly
+	err = ctStatement.QueryRow(historyURL).Scan(&count)
+	th.Assert(t, err == nil, err)
+	th.Assert(t, count == 2, "Should have got 2, got "+strconv.Itoa(count))
+
+	// Call PruneInfoHistory function which will call the history pruning function
+	PruneInfoHistory(ctx, store, false)
+
+	// Info history table should have 2 entries as history pruning will not remove entries less than month old
+	err = ctStatement.QueryRow(historyURL).Scan(&count)
+	th.Assert(t, err == nil, err)
+	th.Assert(t, count == 2, "Should have got 2, got "+strconv.Itoa(count))
+
+	// Clear history table in database
+	_, err = clearStatement.ExecContext(ctx, historyURL)
+	th.Assert(t, err == nil, err)
+	idExpectedArr = nil
+
+	// Add four fhir endpoint info history entries with old entered at date, first and second to last I operations
```

Comment:
I'm concerned about how "real" this scenario is in that there are 2 identical endpoints (they each use testFhirEndpointsInfo) added and updated at the same time twice

Reply:
To clarify -- the testFhirEndpointInfo.URL are identical between the 2 Inserts, I'm not sure we could ever see this given how we treat fhir_endpoints URLs as unique